### ISSUE TITLE
Folder not found

### ISSUE THREAD

**Issue by smartvnm** [sharadcodes/img-resizer#1]
Trying to resize images in assets folder for gh-pages fails, in workflow's log says "find: assets folder can't be found" or similar. Any suggestions, please?

**Comment by sharadcodes** [comment]
Can you please share some code with the project folder structure

**Comment by sharadcodes** [comment]
> Trying to resize images in assets folder for gh-pages fails, in workflow's log says "find: assets folder can't be found" or similar. Any suggestions, please?

Can you please share some code with the project folder structure so that I can get an idea of the problem

**Comment by smartvnm** [comment]
https://github.com/smartvnm/smartvnm.github.io is the repo.
Workflow looks like this
```# This is a basic workflow to help you get started with Actions

name: CI

# Controls when the action will run. Triggers the workflow on push or pull request
# events but only for the master branch
on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      - name: Image Resizer Action
        uses: sharadcodes/img-resizer@v1.0
        with:
          image-folder: 'assets' # This is a Required field & this folder has original images
          new-resolution: '400x200' # This is Required field
          output-folder: "assets/thumbs" # This is Required field & this folder will be used to
```
If it helps, I can paste the whole log, but essentially, on Image Resizer Action job it says
``` find: assets: No such file or directory ```

**Comment by sharadcodes** [comment]
Okay Let me see. 
Till then look at this repo for the usage example.
https://github.com/sharadcodes/resized-images

**Comment by sharadcodes** [comment]
@smartvnm  is it working now ?

**Comment by smartvnm** [comment]
Half way there, I got this line from your example
```- uses: actions/checkout@v2```
It solved the problem with assets folder, it converts the images, but nothing appears in the assets/thumbs folders

**Comment by sharadcodes** [comment]
See you'll have to use the git push in the workflow. Since the resizing action is only for performing conversion & not updating the repo.
You can add something like this
```
    - name: Add & Commit files
      run: |
        git config --local user.email "your-github-username@github.com"
        git config --local user.name "your-github-username"
        git add .
        git commit -m "Add changes" -a
        git push -f -q https://${{ secrets.GITHUB_TOKEN }}@github.com/your-github-username/YOUR-REPO-NAME.git master
```

But this may invoke infinite loop of github workflow because your workflow file has on push event trigger

**Comment by smartvnm** [comment]
While we are at it, I'll throw a few questions here as well:
1. This can be configured to resize at different resolutions at once or it needs to be set up for each resolution? Like, new workflow for new resolution, or I could just add new size under the first size?
2. I've seen somewhere in the code that it checks if images are already resized, I'm (obviously) not a great programmer, so if you could confirm on that, please.
3. More like a suggestion, a little documentation on this wouldn't hurt  :)

**Comment by smartvnm** [comment]
I will try your suggestion and come back with the result. 
If it's true that the resizer checks for already resized images, it shouldn't cause infinite loop as there is nothing to push

**Comment by sharadcodes** [comment]
You don

> While we are at it, I'll throw a few questions here as well:
> 
> 1. This can be configured to resize at different resolutions at once or it needs to be set up for each resolution? Like, new workflow for new resolution, or I could just add new size under the first size?
> 2. I've seen somewhere in the code that it checks if images are already resized, I'm (obviously) not a great programmer, so if you could confirm on that, please.
> 3. More like a suggestion, a little documentation on this wouldn't hurt  :)

**Ans 1**
You don't need to make a new workflow for it, just add one more step in the workflow with the action and you are good to go.
**Ans 2**
Yes it checks for the file in the output folder with the same name if it is there it will not convert it.
**Ans 3**
Thanks for the suggestion :nerd_face: I'll add documentation :notebook: to it. :smile:

**Comment by smartvnm** [comment]
Great, it worked. No infinite loops, as checking for resized images is a very smart thing there. Thanks for help!

PS. When you will build an image compressor too? Your solution here works on push, while what's out there for compressing images works only on pull requests.

**Comment by sharadcodes** [comment]
> Great, it worked. No infinite loops, as checking for resized images is a very smart thing there. Thanks for help!
> 
> PS. When you will build an image compressor too? Your solution here works on push, while what's out there for compressing images works only on pull requests.

You're welcome.
But I don't understant the thing that you asked or explained in your second paragraph

**Comment by smartvnm** [comment]
https://github.com/calibreapp/image-actions
Something like this, but this above (and similar on GitHub) works only on pull requests, not on push. Because they don't check for already compressed images and it would re-compress already compressed images. 
I hope it's clear now :)

**Comment by sharadcodes** [comment]
> https://github.com/calibreapp/image-actions
> Something like this, but this above (and similar on GitHub) works only on pull requests, not on push. Because they don't check for already compressed images and it would re-compress already compressed images.
> I hope it's clear now :)

Ohh I see, so you want any improvements or what ?

**Comment by smartvnm** [comment]
Well, something like that, not necessarily for me, your resizer will help me a lot, I think. But I liked your "check if job already done" idea and it would definitely be an improvement for the image compressor above.

**Comment by sharadcodes** [comment]
> Well, something like that, not necessarily for me, your resizer will help me a lot, I think. But I liked your "check if job already done" idea and it would definitely be an improvement for the image compressor above.

Okay will see and add the feature when possible.
Don't forget to :star: :)
